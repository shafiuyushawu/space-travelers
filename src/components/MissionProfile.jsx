import PropTypes from 'prop-types';

const MissionProfile = ({ member }) => (
  <li className="p-1 text-2xl border">{member.mission_name}</li>
);

MissionProfile.propTypes = {
  member: PropTypes.shape({
    mission_name: PropTypes.string.isRequired,
    member: PropTypes.bool.isRequired,
  }).isRequired,
};
export default MissionProfile;
