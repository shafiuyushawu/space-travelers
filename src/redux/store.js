import { configureStore } from '@reduxjs/toolkit';
import missionSlice from './missions/missionSlice';
import rocketReducer from './rockets/rocketsSlice';

const store = configureStore({
  missions: missionSlice,
  rockets: rocketReducer,
});
export default store;
