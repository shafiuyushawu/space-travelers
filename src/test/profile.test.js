import { render, cleanup } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/store';
import MissionProfile from '../components/MissionProfile';

afterEach(cleanup);

describe('Testing profile components', () => {
  test('Generating the snapshots for MissionProfile', () => {
    const member = {
      mission_name: 'Test Profile',
      member: true,
    };
    const { container } = render(
      <Provider store={store}>
        <MissionProfile member={member} />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
