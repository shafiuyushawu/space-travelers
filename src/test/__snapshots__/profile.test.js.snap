// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div>
  <li
    class="p-1 text-2xl border"
  >
    Test Profile
  </li>
</div>
`;

exports[`Testing profile component Generating the snapshots for MissionProfile 1`] = `
<div>
  <li
    class="p-1 text-2xl border"
  >
    Test Profile
  </li>
</div>
`;
